src/Cryptol/Parser.y

```diff
@@ -298,11 +300,27 @@ mbDoc                   :: { Maybe (Located Text) }
   : doc                    { Just $1 }
   | {- empty -}            { Nothing }
 
+propguards :: { [([Prop PName], Expr PName)] }
+  : 'propguards' '|' propguards_cases { $3 }
+
+propguards_cases :: { [([Prop PName], Expr PName)] }
+  : propguards_case '|' propguards_cases { $1 ++ $3 }
+  | propguards_case { $1 }
+
+propguards_case :: { [([Prop PName], Expr PName)] }
```

Comment:
It might be nicer if we make the `|' part of `propguards_case`, as each of them starts with a `|`.  Then `propguards_cases` would be just a list of 1 or more `propguards_case`, and we wouldn't need the starting `|` in the declaration.

Reply:
I'll see what I can do, this is another place where I ran into difficulties making happy accept the grammer

Reply:
I got frustrated and gave up on it for a while, but I just tried again and it seems to be fixed now! Weird, I'm not sure what I was missing before